doc/CRD-generator.md

```diff
@@ -26,7 +26,215 @@ dependencies {
 Now you can define a `class` that extends `io.fabric8.kubernetes.client.CustomResource`
 and the corresponding CRD is generated in the folder: `target/classes/META-INF/fabric8`
 
-### Supported annotations
+For example, code similar to:
+
+```java
+@Group("org.example")
+@Version("v1alpha1")
+@ShortNames("ex")
+@Plural("examples")
+public class Example extends CustomResource<ExampleSpec, ExampleStatus> implements Namespaced {}
+
+public class ExampleSpec {
+  int someValue;
+}
+
+public class ExampleStatus {
+  boolean error;
+  String message;
+}
+```
+
+Running the `compile` task will generate 2 files
+`target/classes/META-INF/fabric8/examples.org.example-v1.yml` and
+`target/classes/META-INF/fabric8/examples.org.example-v1beta1.yml`,
+the file name is calculated as `<plural>.<group>-<apiVersion>.yml`.
+
+The content of the `v1` looks similar to follow:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: examples.org.example
+spec:
+  group: org.example
+  names:
+    kind: Example
+    plural: examples
+    shortNames:
+    - ex
+    singular: example
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        properties:
+          spec:
+            properties:
+              someValue:
+                type: integer
+            type: object
+          status:
+            properties:
+              error:
+                type: boolean
+              message:
+                type: string
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+```
+
+## Features
+
+### JsonProperty
+
+If a field or one of its accessors is annotated with `JsonProperty`
+
+```java
+public class ExampleSpec {
+  @JsonProperty("myValue")
+  int someValue;
+}
+```
+
+The generated field in the CRD will be named after the value provided in the annotation, such as:
```

Comment:
```suggestion
The generated field in the CRD will be named after the value provided in the annotation instead of using the default name derived from the field name, such as:
```